modules/phase_field/src/materials/EpsAndMAndLAnisoGauss.C

```diff
@@ -0,0 +1,931 @@
+                    (((*_vals[n])[_qp]) * ((*_vals[n])[_qp]));
+
+              // vector of gaussian exponent
+
+              ADReal vsmalbax = (*_grad_vals[m])[_qp](0) - (*_grad_vals[n])[_qp](0);
+              ADReal vsmalbay = (*_grad_vals[m])[_qp](1) - (*_grad_vals[n])[_qp](1);
+              ADReal vsmalbaz = (*_grad_vals[m])[_qp](2) - (*_grad_vals[n])[_qp](2);
+
+              ADReal normvsmallbavalue =
+                  (vsmalbax * vsmalbax) + (vsmalbay * vsmalbay) + (vsmalbaz * vsmalbaz);
+
+              ADReal normvsmallba = std::pow(normvsmallbavalue, .5);
+
+              ADReal vsmallbax = vsmalbax / normvsmallba;
+              ADReal vsmallbay = vsmalbay / normvsmallba;
+              ADReal vsmallbaz = vsmalbaz / normvsmallba;
+
+              _Vx[_qp] = vsmallbax;
+              _Vy[_qp] = vsmallbay;
+              _Vz[_qp] = vsmallbaz;
+
+              if (sumval == 0.0)
+              {
+                vsmallbax = 0.0;
+                vsmallbay = 0.0;
+                vsmallbaz = 0.0;
+                _Vx[_qp] = 0.0;
+                _Vy[_qp] = 0.0;
+                _Vz[_qp] = 0.0;
+              }
+
+              // calculate quaternion qba components
+
+              // Calculation method
+
+              // qa = (a,b,c,d);   qb = (e,f,g,h);  q-1 = (x0, -x1, -x2, -x3);
+
+              // qaqb[w] =  a*e  -  b*f  -  c*g  -  d*h
+              // qaqb[x] =  a*f  +  b*e  +  c*h  -  d*g
+              // qaqb[y] =  a*g  -  b*h  +  c*e  +  d*f
+              // qaqb[z] =  a*h  +  b*g  -  c*f  +  d*e
+
+              //***************** qba = qa-1 * qb
+              //****************************************************** a = qwR[a] b = qxR[a] * (-1)
+              // c = qyR[a] * (-1)
+              // d = qzR[a] * (-1)
+
+              // e = qwR[b]
+              // f = qxR[b]
+              // g = qyR[b]
+              // h = qzR[b]
+
+              ADReal qbaw = (qwR[a] * qwR[b]) - (qxR[a] * (-1) * qxR[b]) -
+                            (qyR[a] * (-1) * qyR[b]) - (qzR[a] * (-1) * qzR[b]);
+              ADReal qbax = (qwR[a] * qxR[b]) + (qxR[a] * (-1) * qwR[b]) +
+                            (qyR[a] * (-1) * qzR[b]) - (qzR[a] * (-1) * qyR[b]);
+              ADReal qbay = (qwR[a] * qyR[b]) - (qxR[a] * (-1) * qzR[b]) +
+                            (qyR[a] * (-1) * qwR[b]) + (qzR[a] * (-1) * qxR[b]);
+              ADReal qbaz = (qwR[a] * qzR[b]) + (qxR[a] * (-1) * qyR[b]) -
+                            (qyR[a] * (-1) * qxR[b]) + (qzR[a] * (-1) * qwR[b]);
+
+              // convert anisotropy quaternion qba to vector vba and thetaba
+              ADReal thetaba = 2 * (std::acos(qbaw));
+
+              ADReal vbaxori = qbax / (std::sin((thetaba) / 2));
+              ADReal vbayori = qbay / (std::sin((thetaba) / 2));
+              ADReal vbazori = qbaz / (std::sin((thetaba) / 2));
+
+              ADReal rrrvalue = (vbaxori * vbaxori) + (vbayori * vbayori) + (vbazori * vbazori);
```

Comment:
```suggestion
              ADReal rrrvalue = vbaxori * vbaxori + vbayori * vbayori + vbazori * vbazori;
```

You can rely on operator precedence!

Reply:
That was fixed. See updated files.